netaddr.go

```diff
@@ -122,13 +122,24 @@ func IPv4(a, b, c, d uint8) IP {
 // s can be in dotted decimal ("192.0.2.1"), IPv6 ("2001:db8::68"),
 // or IPv6 with a scoped addressing zone ("fe80::1cc0:3e8c:119f:c2e1%ens18").
 func ParseIP(s string) (IP, error) {
-	// TODO: do our own parsing to save some allocs? For now,
-	// while showing off new API & representation, just use the
-	// standard library's parsing.
-	ipa, err := net.ResolveIPAddr("ip", s)
-	if err != nil {
-		return IP{}, err
+	var ipa net.IPAddr
+	ipa.IP = net.ParseIP(s)
+	if ipa.IP == nil {
+		// net.ParseIP can't deal with zoned scopes, lets split and try to parse the ip again
+		if index := strings.Index(s, "%"); index > -1 {
```

Comment:
I'd name this variable either `i` (which is the conventional `index` name), or something more meaningful like `percent`.

Also, `!= -1` is more common.

```go
if percent := strings.Index(s, "%"); percent != -1 {
      s, ipa.Zone = s[:percent], s[percent+1:]
      if ipa.Zone == "" {
           return IP{}, fmt.Errorf("netaddr.ParseIP(%q): missing zone", s)
      }
      ipa.IP = net.ParseIP(s)
}
if ipa.IP == nil {
	return IP{}, fmt.Errorf("netaddr.ParseIP(%q): unable to parse IP", s)
}
```

Reply:
fixed